bin/asgs-lint

```diff
@@ -62,17 +62,39 @@ if [ $(egrep -c "^ *HINDCASTLENGTH *= *[1-9][0-9]*.[0-9]+ *$" $ASGS_CONFIG) != 1
 fi
 
 ## 20
-### check BACKGROUNDMET and TROPICALCYCLONE, fatal if both are 'on'; warn if one is NOT 'on' 
+### check BACKGROUNDMET and TROPICALCYCLONE, fatal if both are 'on'; warn if one is NOT 'on'
 if [[ "$BACKGROUNDMET" == "on" && "$TROPICALCYCLONE" == "on" ]]; then
   fatal "BACKGROUNDMET and TROPICALCYCLONE can't both be 'on'. One must be 'on', the other must be 'off'."
 fi
 
+### check formatting of COLDSTARTDATE and make sure it has valid values
+### ref. https://github.com/StormSurgeLive/asgs/issues/360
+fatalMsg=$(
```

Comment:
I need to fix the way we extract values statically. Is `COLDSTARTDATE` being extracted?

Regarding the check, this check doesn't seem to be scaling well. I mean I don't want to force a challenge to do these kinds of things in the most difficult way possible, so I need to think about this.

My current thinking is to "warn" if it's not using `$(get-coldstart-date)` rather than validate the actual date value - this seems like this is more appropriately handled either in the running code - or - we start creating helper check scripts.

Reply:
Yeah this is not an optimal way to check COLDSTARTDATE, I am just playing with ideas really. The question is: in what context will the linter be executed? As ASGS is running? Inside `asgsh` but not when the ASGS is running? Outside `asgsh`? Not sure we've established this.

Reply:
Also, is it checking just the config file? Or also the environment variables specified in the profile?

Reply:
I always have them in my config files. I am thinking about redoing this in Perl as just as an experiment to see if it scales better - I think it should.

Reply:
And to be clear, it's nothing you did. Just seeing how it could get hair quickly and that `egrep` regex I had to use for `BACKGROUNDMET` makes me think this is not going to be a good case for `bash`. It could be jsimilarly structured in Perl.

Reply:
Oh, sorry missed a question - right now it's just checking ``$ASGS_CONFIG`.

Reply:
After some more thinking, the way to quantify a "linter" is that it checks for things that can be checked without actually executing any code. So the "dumber" the "better". In our case, the linter is really meant to do some high level passes but checking things like a validate date I think is reasonable.  Don't hold me to this, but I think if we can detect something in the ENVIRONMENT+`$ASGS_CONFIG` that will cause `asgs_main.sh` to either fail immediately, or cause a situation that is not discovered until hours later (say during an operational run) _and_ it can be determined _statically_ then we should add it. I have an idea for a self-contained Perl script that will allow us to add things and keep the checks rightly ordered. The checking of the date alone is reason to switch to Perl. So I will whip up a perl based linter and add it to this PR just so we can compare.

Reply:
Ok sounds good, thank you for the clarification.